### PR TITLE
[luci] Support MatrixDiag operator

### PR DESCRIPTION
**Pull request by d-krylov**
Support MatrixDiag operator in luci
    
ONE-DCO-1.0-Signed-off-by: Denis Krylov <krylovdenism@gmail.com>

### CODE REVIEW

File: compiler/luci/export/src/CircleOperationExporter.cpp [hunk 1]
```diff
@@ -629,6 +630,19 @@ void OperationExporter::visit(luci::CircleLogSoftmax *node)
...
+{
+  uint32_t op_idx = md.registerBuiltinOpcode(circle::BuiltinOperator_MATRIX_DIAG);
+  std::vector<int32_t> inputs_vec{get_tensor_index(node->diagonal())};
+  std::vector<int32_t> outputs_vec{get_tensor_index(static_cast<loco::Node *>(node))};
+  auto inputs = builder.CreateVector(inputs_vec);
+  auto outputs = builder.CreateVector(outputs_vec);
+  auto options = CreateMatrixDiagOptions(builder);
+  auto op_offset = CreateOperator(builder, op_idx, inputs, outputs,
+                                  circle::BuiltinOptions_MatrixDiagOptions, options.Union());
+  gd._operators.push_back(op_offset);
```
Comment (seanshpark) [inline compiler/luci/export/src/CircleOperationExporter.cpp:643]: Linke `MatrixSetDiag` can you also use `export_simple` here?

File: compiler/luci/tests/test.lst [hunk 2]
```diff
@@ -76,6 +76,7 @@ addread(LogicalOr_000)
 addread(Logistic_000)
 addread(LogSoftmax_000)
 addread(MatMul_000)
+addread(MatrixDiag_000)
```
Comment (binarman) [inline compiler/luci/tests/test.lst:79]: Will it work with this tests enabled?

Comment (d-krylov) [reply compiler/luci/tests/test.lst:79]: Yes. 
```
./nncc build
```
works.

**Comment by seanshpark** [comment]
Test was removed but this PR looks OK :)

**Pull request merged**